tutorials/e2e-huggingface-summarization/e2e-dl-huggingface-summarization.ipynb

```diff
@@ -0,0 +1,1103 @@
+    "cluster_basic = ml_client.begin_create_or_update(cluster_basic)\n",
+    "\n",
+    "print(\n",
+    "    f\"AMLCompute with name {cluster_basic.name} is created, the compute size is {cluster_basic.size}\"\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12ccf792",
+   "metadata": {},
+   "source": [
+    "# 1. Preparing the Resources\n",
+    "\n",
+    "## 1.1. Create a Job Environment\n",
+    "So far, in the requirements section, we have created a development environment on our development machine. AML needs to know what environment to use for each step of the pipeline. We can use any published docker image as is, or add or required dependencies to the image.In this example, we create a conda environment for our jobs, using a [conda yaml file](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually) and add it to an Ubuntu image in Microsoft Container Registry. For more information on AML environments and Azure Container Registries, please check [AML documentaiton](https://docs.microsoft.com/en-us/azure/machine-learning/concept-environments).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "02858031",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import Environment\n",
+    "import os\n",
+    "\n",
+    "custom_env_name = \"transformers-gpu\" if USE_GPU else \"transformers-cpu\"\n",
+    "\n",
+    "pipeline_job_env = Environment(\n",
+    "    name=custom_env_name,\n",
+    "    description=\"Custom environment for transformer model training\",\n",
+    "    tags={\"transformers\": \">4.11.0\"},\n",
+    "    conda_file=os.path.join(\"dependencies\", \"transformers_conda.yml\"),\n",
+    "    image=\"mcr.microsoft.com/azureml/openmpi4.1.0-cuda11.1-cudnn8-ubuntu18.04\"\n",
+    "    if USE_GPU\n",
+    "    else \"mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04:20220218.v1\",\n",
+    ")\n",
+    "pipeline_job_env = ml_client.environments.create_or_update(pipeline_job_env)\n",
+    "\n",
+    "print(\n",
+    "    f\"Environment with name {pipeline_job_env.name} is created, the version is {pipeline_job_env.version}\"\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34e89b69",
+   "metadata": {},
+   "source": [
+    "## 1.2. Create or Load Components\n",
+    "Now that we have our workspace, compute and input data ready, let's work on the individual steps of our pipeline. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "79cd41a3",
+   "metadata": {},
+   "source": [
+    "### 1.2.1 Baseline Evaluation Component\n",
+    "Text summarization metrics are not standardized, therefore it is a good practice if we create a baseline to compare the results we get from our transformer models. A simple method can be using the first three sentences of the text as the summary. We have implemented that in the baseline component below. "
```

Comment:
I feel there's a gap here, it's hard to understand where the pipeline is going. I think the explanation might be too short. Or maybe providing some high level structure at the beginning of the notebook in the motivations section could help?

Reply:
I'd suggest explaining how summarization works and what's the goal of the pipeline under the motivations section.

Reply:
done